client/my-sites/menus/menu-editable-item.jsx

```diff
@@ -164,6 +171,20 @@ var MenuEditableItem = React.createClass( {
 				type_family: { $set: this.state.itemType.family }
 			}
 		} ) );
+
+		if (
+			!this.state.userChangedName &&
+			this.isNew() &&
+			this.isMounted() &&
+			has( this, 'refs.menuLabel' )
+		) {
+			if ( isInjectedNewPageItem( content ) ) {
+				this.refs.menuLabel.focus();
+				this.setState( { placeholderText: this.translate( 'Enter page name', { textOnly: true } ) } );
+			} else {
+				this.setState( { placeholderText: this.translate( 'Enter menu label', { textOnly: true } ) } );
+			}
+		}
```

Comment:
I also feel bad about it!
My heart was hurting while writing that!

Code reuse was impossible at the time of writing this PR, because there was even more complexity to it :)
we can follow it up though!